php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1702 @@
+      arrayToken,
+      openParenthesis,
+      elements.isPresent() ? elements.get() : new SeparatedList<>(ImmutableList.<ArrayPairTree>of(), ImmutableList.<InternalSyntaxToken>of()),
+      closeParenthesis);
+  }
+
+  public ArrayInitializerTree newArrayInitBracket(InternalSyntaxToken openBracket, Optional<SeparatedList<ArrayPairTree>> elements, InternalSyntaxToken closeBracket) {
+    return new ArrayInitializerBracketTreeImpl(
+      openBracket,
+      elements.isPresent() ? elements.get() : new SeparatedList<>(ImmutableList.<ArrayPairTree>of(), ImmutableList.<InternalSyntaxToken>of()),
+      closeBracket);
+  }
+
+  public FunctionExpressionTree functionExpression(Optional<InternalSyntaxToken> staticToken, InternalSyntaxToken functionToken, Optional<InternalSyntaxToken> ampersandToken,
+    ParameterListTree parameters, Optional<LexicalVariablesTree> lexicalVariables, BlockTree block) {
+
+    return new FunctionExpressionTreeImpl(
+      staticToken.orNull(),
+      functionToken,
+      ampersandToken.orNull(),
+      parameters,
+      lexicalVariables.orNull(),
+      block);
+  }
+
+  public NewExpressionTree newExpression(InternalSyntaxToken newToken, ExpressionTree expression) {
+    return new NewExpressionTreeImpl(newToken, expression);
+  }
+
+  public ExitTreeImpl newExitExpression(InternalSyntaxToken openParenthesis, Optional<ExpressionTree> expressionTreeOptional, InternalSyntaxToken closeParenthesis) {
+    return new ExitTreeImpl(openParenthesis, expressionTreeOptional.orNull(), closeParenthesis);
+  }
+
+  public ExitTree completeExitExpression(InternalSyntaxToken exitOrDie, Optional<ExitTreeImpl> partial) {
+    return partial.isPresent() ? partial.get().complete(exitOrDie) : new ExitTreeImpl(exitOrDie);
+  }
+
+  public ExpressionTree combinedScalarOffset(ArrayInitializerTree arrayInitialiser, Optional<List<ArrayAccessTree>> offsets) {
+    ExpressionTree result = arrayInitialiser;
+    for (ArrayAccessTree offset : optionalList(offsets)) {
+      result = ((ArrayAccessTreeImpl) offset).complete(result);
+    }
+
+    return result;
+  }
+
+  public ExpressionTree postfixExpression(ExpressionTree expression, Optional<Object> optional) {
+    if (optional.isPresent()) {
+
+      if (optional.get() instanceof SyntaxToken) {
+        SyntaxToken operator = ((SyntaxToken) optional.get());
+
+        return new PostfixExpressionTreeImpl(
+          operator.text().equals(PHPPunctuator.INC.getValue()) ? Kind.POSTFIX_INCREMENT : Kind.POSTFIX_DECREMENT,
+          expression,
+          operator);
+
+      } else {
+        Tuple<InternalSyntaxToken, ExpressionTree> tuple = ((Tuple) optional.get());
+        return new BinaryExpressionTreeImpl(Kind.INSTANCE_OF, expression, tuple.first(), tuple.second);
+      }
+    }
+
+    return expression;
+  }
+
+  public AssignmentExpressionTree assignmentExpression(ExpressionTree lhs, InternalSyntaxToken operatorToken, ExpressionTree rhs) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "assignmentExpression" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)